src/http.cc

```diff
@@ -1910,11 +2026,19 @@ HttpStateData::httpBuildRequestHeader(HttpRequest * request,
         delete cc;
     }
 
-    /* maybe append Connection: keep-alive */
-    if (flags.keepalive) {
-        hdr_out->putStr(Http::HdrType::CONNECTION, "keep-alive");
+    if (hdr_in->has(Http::HdrType::UPGRADE)) {
+        String upgradeValue = hdr_in->getList(Http::HdrType::UPGRADE);
+        const char *upGrdValue = mkUpgradeHeader(state, request, upgradeValue);
+        if (upGrdValue && *upGrdValue) {
```

Comment:
Yes but then we would have to convert it again to "const char *" using the costly (because reallocates) SBuf::c_str to use it with HttpHeaderEntry.
My opinion is that converting to SBuf should be resolved with other similar cases when the HttpMsg start using SBuf instead of String.